src/docfx/lib/markdown/validation/ContentValidationContext.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Validations.DocFx.Adapter;
+
+namespace Microsoft.Docs.Build
+{
+    internal class ContentValidationContext : IValidationContext
+    {
+        private const string ValidationApiBaseUrl = "validationApiBaseUrl";
+        private const string DefaultValidationApiBaseUrl = "https://docs.microsoft.com";
+        private const string ValidationRequestTimeoutInSeconds = "validationRequestTimeoutInSeconds";
+        private const int DefaultValidationRequestTimeoutInSeconds = 10;
+        private const string ValidationMaxTries = "validationMaxTries";
+        private const int DefaultValidationMaxTries = 4;
+        private const string ValidationRetryDelaySeconds = "validationRetryDelaySeconds";
+        private const int DefaultValidationRetryDelaySeconds = 3;
+        private const string ValidationMaxTransientFailureCount = "validationMaxTransientFailureCount";
+        private const int DefaultValidationMaxTransientFailureCount = 20;
+
+        public string MarkdownRulesFilePath { get; set; }
+
+        public string MetadataRulesFilePath { get; set; }
+
+        public string RepositoryUrl { get; set; }
+
+        public string Branch { get; set; }
+
+        public string ApiBase { get; set; }
+
+        public TimeSpan RequestTimeout { get; set; }
+
+        public int MaxTries { get; set; }
+
+        public int RetryDelaySeconds { get; set; }
+
+        public int MaxTransientFailureCount { get; set; }
+
+        public ContentValidationContext(Config config)
+        {
+            MarkdownRulesFilePath = config.ContentValidationRulesFilePath;
+            RepositoryUrl = EnvironmentVariable.RepositoryUrl;
+            Branch = EnvironmentVariable.RepositoryBranch;
+            ApiBase = config.GlobalMetadata.TryGetValue(ValidationApiBaseUrl, out var apiBase) ? apiBase.ToObject<string>() : DefaultValidationApiBaseUrl;
```

Comment:
_global metadata_ is for user configuration, these are systems settings, consider hard code it here.

Reply:
make sense

Reply:
This value is different in PPE and PROD. Instead of hard code, we need a way to pass system settings.

Reply:
@LowEntropyBody , is this URL for metadata validation only? markdown validation should never call external api, thus how about leave this setting and other network call related settings to default so it does not confuse people?